src/components/expiryPicker.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import "./expiryPicker.scss";
+
+import * as React from "react";
+
+import * as moment from "moment";
+import * as mask from "text-mask-core";
+
+import { Checkbox } from "azure-devops-ui/Checkbox";
+import { ObservableValue } from "azure-devops-ui/Core/Observable";
+import { FormItem } from "azure-devops-ui/FormItem";
+import { TextField } from "azure-devops-ui/TextField";
+
+interface IExpiryPickerProps {
+    expiryDate: Date;
+    onChange?(date: Date): void;
+    onParseError?(message: string): void;
+}
+
+interface IExpiryPickerState {
+    expiryEnabled: boolean;
+    dateError: boolean;
+}
+
+export class ExpiryPicker extends React.Component<IExpiryPickerProps, IExpiryPickerState> {
+    private textMaskInputElement: any;
+    private dateStringObservable: ObservableValue<string> = new ObservableValue<string>("");
+
+    constructor(props: IExpiryPickerProps) {
+        super(props);
+
+        this.state = {
+            expiryEnabled: props.expiryDate != null,
+            dateError: false,
+        };
+    }
+
+    public componentDidMount(): void {
+        this.setupFields();
+    }
+
+    public componentDidUpdate(prevProps: IExpiryPickerProps): void {
+        if (JSON.stringify(prevProps.expiryDate) !== JSON.stringify(this.props.expiryDate)) {

```

Comment:
Need to stringify here?  Maybe direct compare (or getTime() comapre)?

Reply:
+1